diff -up bash-completion-1.90/bash_completion~ bash-completion-1.90/bash_completion
--- bash-completion-1.90/bash_completion~	2011-11-03 11:51:35.000000000 +0200
+++ bash-completion-1.90/bash_completion	2011-11-04 21:34:14.073711442 +0200
@@ -45,7 +45,7 @@ readonly BASH_COMPLETION_COMPAT_DIR
 
 # Blacklisted completions, causing problems with our code.
 #
-_blacklist_glob='@(acroread.sh)'
+_blacklist_glob='@()'
 
 # Turn on extended globbing and programmable completion
 shopt -s extglob progcomp
@@ -1878,7 +1878,7 @@ if [[ -d $BASH_COMPLETION_COMPAT_DIR && 
             && -f $i && -r $i ]] && . "$i"
     done
 fi
-unset i
+unset i _blacklist_glob
 
 # source user completion file
 [[ ${BASH_SOURCE[0]} != ~/.bash_completion && -r ~/.bash_completion ]] \
