From 3a205c536c3c811b393d635799d19cf9209ca84a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ville=20Skytt=C3=A4?= <ville.skytta@iki.fi>
Date: Tue, 3 Nov 2009 23:36:01 +0200
Subject: [PATCH] Fix service argument completion.

---
 bash_completion |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/bash_completion b/bash_completion
index e1c4054..6f1978c 100644
--- a/bash_completion
+++ b/bash_completion
@@ -757,7 +757,7 @@ deinstall clean clean-depends kernel buildworld' make
             _services
         else
             COMPREPLY=( $( compgen -W '`sed -ne "y/|/ /; \
-            s/^.*\(U\|msg_u\)sage.*{\(.*\)}.*$/\1/p" \
+            s/^.*\(U\|msg_u\)sage.*{\(.*\)}.*$/\2/p" \
             $sysvdir/${prev##*/} 2>/dev/null`' -- "$cur" ) )
         fi
 
-- 
1.7.0

