diff -up bash_completion/bash_completion~ bash_completion/bash_completion
--- bash_completion/bash_completion~	2006-03-01 18:20:18.000000000 +0200
+++ bash_completion/bash_completion	2008-09-07 12:12:11.000000000 +0300
@@ -348,10 +348,10 @@ _services()
 	local sysvdir famdir
 	[ -d /etc/rc.d/init.d ] && sysvdir=/etc/rc.d/init.d || sysvdir=/etc/init.d
 	famdir=/etc/xinetd.d
-	COMPREPLY=( $( builtin echo $sysvdir/!(*.rpmsave|*.rpmorig|*~|functions)) )
+	COMPREPLY=( $( builtin echo $sysvdir/!(*.rpmsave|*.rpmorig|*.rpmnew|*~|functions)) )
 
 	if [ -d $famdir ]; then
-		COMPREPLY=( ${COMPREPLY[@]} $( builtin echo $famdir/!(*.rpmsave|*.rpmorig|*~)) )
+		COMPREPLY=( ${COMPREPLY[@]} $( builtin echo $famdir/!(*.rpmsave|*.rpmorig|*.rpmnew|*~)) )
 	fi
 
 	COMPREPLY=( $( compgen -W '${COMPREPLY[@]#@($sysvdir|$famdir)/}' -- $cur ) )
@@ -8679,7 +8679,7 @@ _update_rc_d()
     [ -d /etc/rc.d/init.d ] && sysvdir=/etc/rc.d/init.d \
 	|| sysvdir=/etc/init.d
 
-    services=( $(echo $sysvdir/!(README*|*.sh|*.dpkg*|*.rpm*)) )
+    services=( $(echo $sysvdir/!(README*|*.sh|*.dpkg*|*.rpmsave|*.rpmorig|*.rpmnew)) )
     services=( ${services[@]#$sysvdir/} )
     options=( -f -n )
 
@@ -8745,7 +8745,7 @@ _invoke_rc_d()
     [ -d /etc/rc.d/init.d ] && sysvdir=/etc/rc.d/init.d \
 	|| sysvdir=/etc/init.d
 
-    services=( $(echo $sysvdir/!(README*|*.sh|*.dpkg*|*.rpm*)) )
+    services=( $(echo $sysvdir/!(README*|*.sh|*.dpkg*|*.rpmsave|*.rpmorig|*.rpmnew)) )
     services=( ${services[@]#$sysvdir/} )
     options=( --help --quiet --force --try-anyway --disclose-deny --query --no-fallback )
 
@@ -9281,7 +9281,7 @@ unset list
 if [ -d $BASH_COMPLETION_DIR -a -r $BASH_COMPLETION_DIR -a \
      -x $BASH_COMPLETION_DIR ]; then
 	for i in $BASH_COMPLETION_DIR/*; do
-		[[ ${i##*/} != @(*~|*.bak|*.swp|\#*\#|*.dpkg*|.rpm*) ]] &&
+		[[ ${i##*/} != @(*~|*.bak|*.swp|\#*\#|*.dpkg*|*.rpmsave|*.rpmorig|*.rpmnew) ]] &&
 			[ \( -f $i -o -h $i \) -a -r $i ] && . $i
 	done
 fi
