--- bash_completion~	2009-01-06 19:12:03.000000000 +0200
+++ bash_completion	2009-01-06 19:23:35.000000000 +0200
@@ -424,10 +424,10 @@
 	local sysvdir famdir
 	[ -d /etc/rc.d/init.d ] && sysvdir=/etc/rc.d/init.d || sysvdir=/etc/init.d
 	famdir=/etc/xinetd.d
-	COMPREPLY=( $( builtin echo $sysvdir/!(*.rpmsave|*.rpmorig|*~|functions)) )
+	COMPREPLY=( $( builtin echo $sysvdir/!(*.rpmsave|*.rpmorig|*.rpmnew|*~|functions)) )
 
 	if [ -d $famdir ]; then
-		COMPREPLY=( "${COMPREPLY[@]}" $( builtin echo $famdir/!(*.rpmsave|*.rpmorig|*~)) )
+		COMPREPLY=( "${COMPREPLY[@]}" $( builtin echo $famdir/!(*.rpmsave|*.rpmorig|*.rpmnew|*~)) )
 	fi
 
 	COMPREPLY=( $( compgen -W '${COMPREPLY[@]#@($sysvdir|$famdir)/}' -- $cur ) )
@@ -9080,7 +9080,7 @@
     [ -d /etc/rc.d/init.d ] && sysvdir=/etc/rc.d/init.d \
 	|| sysvdir=/etc/init.d
 
-    services=( $(echo $sysvdir/!(README*|*.sh|*.dpkg*|*.rpm*)) )
+    services=( $(echo $sysvdir/!(README*|*.sh|*.dpkg*|*.rpmsave|*.rpmorig|*.rpmnew)) )
     services=( ${services[@]#$sysvdir/} )
     options=( -f -n )
 
@@ -9146,7 +9146,7 @@
     [ -d /etc/rc.d/init.d ] && sysvdir=/etc/rc.d/init.d \
 	|| sysvdir=/etc/init.d
 
-    services=( $(echo $sysvdir/!(README*|*.sh|*.dpkg*|*.rpm*)) )
+    services=( $(echo $sysvdir/!(README*|*.sh|*.dpkg*|*.rpmsave|*.rpmorig|*.rpmnew)) )
     services=( ${services[@]#$sysvdir/} )
     options=( --help --quiet --force --try-anyway --disclose-deny --query --no-fallback )
 
@@ -9700,7 +9700,7 @@
 if [ -d $BASH_COMPLETION_DIR -a -r $BASH_COMPLETION_DIR -a \
      -x $BASH_COMPLETION_DIR ]; then
 	for i in $BASH_COMPLETION_DIR/*; do
-		[[ ${i##*/} != @(*~|*.bak|*.swp|\#*\#|*.dpkg*|.rpm*) ]] &&
+		[[ ${i##*/} != @(*~|*.bak|*.swp|\#*\#|*.dpkg*|*.rpmsave|*.rpmorig|*.rpmnew) ]] &&
 			[ \( -f $i -o -h $i \) -a -r $i ] && . $i
 	done
 fi
