--- bash_completion~	2006-03-01 18:20:18.000000000 +0200
+++ bash_completion	2006-12-26 14:01:59.000000000 +0200
@@ -2589,6 +2589,7 @@
 		COMPREPLY=( $( ssh -o 'Batchmode yes' $userhost \
 			       command ls -aF1d "$path*" 2>/dev/null | \
 			       sed -e 's/[][(){}<>",:;^&!$&=?`|\ ]/\\\\\\&/g' \
+				   -e "s/'/\\\\\\\\\\\\&/g" \
 				   -e 's/[*@|=]$//g' -e 's/[^\/]$/& /g' ) )
 		return 0
 	fi
@@ -2597,6 +2598,7 @@
 		COMPREPLY=( ${COMPREPLY[@]} $( command ls -aF1d $cur* \
 			    2>/dev/null | sed \
 			    -e 's/[][(){}<>",:;^&!$&=?`|\ ]/\\&/g' \
+			    -e "s/'/\\\\&/g" \
 			    -e 's/[*@|=]$//g' -e 's/[^\/]$/& /g' ) )
 	return 0
 }
