diff -up bash_completion/bash_completion~ bash_completion/bash_completion
--- bash_completion/bash_completion~	2006-03-01 18:20:18.000000000 +0200
+++ bash_completion/bash_completion	2008-09-11 01:29:49.000000000 +0300
@@ -3281,7 +3281,7 @@ complete -F _mysqladmin mysqladmin
 have gzip &&
 _gzip()
 {
-	local cur prev xspec IFS=$'\t\n'
+	local cur prev xspec
 
 	COMPREPLY=()
 	cur=${COMP_WORDS[COMP_CWORD]}
@@ -3290,7 +3290,7 @@ _gzip()
 			--stdout --decompress --force --help --list \
 			--license --no-name --name --quiet --recursive \
 			--suffix --test --verbose --version --fast \
-			--best' -- $cur ) )
+			--best --rsyncable' -- $cur ) )
 		return 0
 	fi
 
